specification/ml/_types/inference.ts

```diff
@@ -0,0 +1,364 @@
+   */
+  classification_labels: string[]
+  /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
+  results_field?: string
+  /** Indicates if more than one true label exists.
+   * @server_default false
+   **/
+  multi_label?: boolean
+  /** The labels to predict. */
+  labels?: string[]
+}
+
+/** Pass through configuration options */
+export class PassThroughInferenceOptions {
+  /** The tokenization options */
+  tokenization?: TokenizationConfigContainer
+  /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
+  results_field?: string
+}
+
+/** Text embedding inference options */
+export class TextEmbeddingInferenceOptions {
+  /** The tokenization options */
+  tokenization?: TokenizationConfigContainer
+  /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
+  results_field?: string
+}
+
+/** Named entity recognition options */
+export class NerInferenceOptions {
+  /** The tokenization options */
+  tokenization?: TokenizationConfigContainer
+  /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
+  results_field?: string
+  /** The token classification labels. Must be IOB formatted tags */
+  classification_labels?: string[]
+}
+
+/** Fill mask inference options */
+export class FillMaskInferenceOptions {
+  /** Specifies the number of top class predictions to return. Defaults to 0. */
+  num_top_classes?: integer
+  /** The tokenization options to update when inferring */
+  tokenization?: TokenizationConfigContainer
+  /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
+  results_field?: string
+}
+
+// Update containers that may only allow a subset of the options
+
+export class NlpInferenceConfigUpdateContainer {
+  /** Text classification configuration for inference. */
+  text_classification?: TextClassificationInferenceUpdateOptions
+  /** Zeroshot classification configuration for inference. */
+  zero_shot_classification?: ZeroShotClassificationInferenceUpdateOptions
+  /** Fill mask configuration for inference. */
+  fill_mask?: FillMaskInferenceUpdateOptions
+  /** Named entity recognition configuration for inference. */
+  ner?: NerInferenceUpdateOptions
+  /** Pass through configuration for inference. */
+  pass_through?: PassThroughInferenceUpdateOptions
+  /** Text embedding configuration for inference. */
+  text_embedding?: TextEmbeddingInferenceUpdateOptions
+}
+
+export class PipelineAggInferenceConfigUpdateContainer {
+  /** Regression configuration for inference. */
+  regression?: RegressionInferenceUpdateOptions
+  /** Classification configuration for inference. */
+  classification?: ClassificationInferenceUpdateOptions
+}
+
+export class RegressionInferenceUpdateOptions {
```

Comment:
The `<ModelType>UpdateOptions` classes look similar to the `<ModelType>Options` classes, is there a reason we need both?

Reply:
For NLP, it is definitely needed as the tokenization options are different (its a subset). 

For regression and classification, I can make them the same object